tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java

```diff
@@ -27,20 +38,80 @@ public void run() {
-
-            final var response = String.join("\r\n",
-                    "HTTP/1.1 200 OK ",
-                    "Content-Type: text/html;charset=utf-8 ",
-                    "Content-Length: " + responseBody.getBytes().length + " ",
-                    "",
-                    responseBody);
+            if (!request.isGetMethod()) {
+                throw new IllegalStateException("아직 지원하지 않는 http 요청입니다.");
+            }
+            final var response = doGet(request);
 
             outputStream.write(response.getBytes());
             outputStream.flush();
         } catch (IOException | UncheckedServletException e) {
             log.error(e.getMessage(), e);
         }
     }
+
+    private Request extractRequest(final InputStream inputStream) throws IOException {
+        final var bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+
+        final var requestMessage = new StringBuilder();
+
+        while (true) {
+            final var buffer = bufferedReader.readLine();
+            requestMessage.append(buffer)
+                    .append("\r\n");
+            if (buffer == null || buffer.length() == 0) {
+                break;
+            }
+        }
+        return Request.of(requestMessage.toString());
+    }
+
+    private Response doGet(final Request request) throws IOException {
+        if (request.isFileRequest()) {
+            return getResponseWithFileName(request.getUri().getResourcePath());
+        }
+        if (request.hasQueryString()) {
+            return getResponseWithQueryString(request);
+        }
+        return getResponse(request);
+    }
+
+    private Response getResponse(final Request request) throws IOException {
+        if (request.isMatchUri("/login")) {
+            return getResponseWithFileName("/login.html");
+        }
+        if (request.isMatchUri("/")) {
+            return Response.okWithResponseBody(ContentType.HTML, "Hello world!");
+        }
+        return new Response();
+    }
+
+    private Response getResponseWithFileName(final String fileName) throws IOException {
+        final var fileExtension = fileName.split("\\.")[1];
+
+        final var contentType = ContentType.from(fileExtension);
+        final var resource = getClass().getClassLoader().getResource("static" + fileName);
+        final var path = new File(resource.getPath()).toPath();
+
+        final var responseBody = new String(Files.readAllBytes(path));
+
+        return Response.okWithResponseBody(contentType, responseBody);
+    }
+
+    private Response getResponseWithQueryString(final Request request) {
+        if (request.isMatchUri("/login")) {
+            final QueryStrings queryStrings = request.getUri().getQueryStrings();
+            final Optional<User> user = InMemoryUserRepository.findByAccount(queryStrings.getValue("account"));
+
+            if (user.isPresent() && user.get().checkPassword(queryStrings.getValue("password"))) {
```

Comment:
ㅎㅎ 이부분
```suggestion
            if (user가_존재하는가(user)) {
```
이렇게 바꿔볼 수 있을 것 같아요!

Reply:
함수로 추출하는 것이 좋겠군요!
InMemoryUserRepository에서 존재하는지 확인 후, 값을 가져오는걸로 리팩토링했습니다. 감사합니다 😄